src/tests/modules/serializer_deserializer.rs

```diff
@@ -540,3 +540,20 @@ fn test_serialize_deserialize_document() {
     let bad_point: Result<Point, crate::de::Error> = from_document(bad_point);
     assert!(bad_point.is_err());
 }
+
+#[test]
+fn test_deserialize_invalid_array_length_issue243() {
+    let _guard = LOCK.run_concurrently();
+    let buffer = b"\n\x00\x00\x00\x04\x00\x00\x00\x00\x00";
+    let _ = Document::from_reader(&mut std::io::Cursor::new(buffer)); 
```

Comment:
Here and below, can we add an `expect_err` assertion to ensure we're rejecting the invalid inputs?